src/buildtools/AssemblyCheck/AssemblyCheck.fs

```diff
@@ -79,6 +79,8 @@ module AssemblyCheck =
             |> List.filter (fun p -> not (p.Contains(@"\FSharpSdk\")))
             |> List.filter (fun a ->
                 let fileProductVersion = FileVersionInfo.GetVersionInfo(a).ProductVersion
+                if (isNull fileProductVersion) then
+                    printfn "Null product version: %s" a
```

Comment:
Given that a `null` string is so bad here can I suggest that `fileProductVersion` be defaulted to an empty string so that the regex match can fail appropriately?  That way if there are multiple issues we'll find them all at once rather than one at a time.

``` diff
-let fileProductVersion = FileVersionInfo.GetVersionInfo(a).ProductVersion
+let mutable fileProductVersion = FileVersionInfo.GetVersionInfo(a).ProductVersion
 if (isNull fileProductVersion) then
-    printfn "Null product version: %s" a
+    fileProductVersion <- String.Empty
```